gap/DocumentationTree.gi

```diff
@@ -178,22 +186,26 @@ InstallMethod( DocumentationTree, [ ],
 end );
 
 ## create a chapter, section or subsection
-InstallMethod( StructurePartInTree, [ IsTreeForDocumentation, IsList ],
+## This first helper, given tree and context, returns a list of two entries.
+## The second entry is a node to represent the structure part at that context.
+## The first is true if that structure part did not yet exist, in which case the
+## node returned as the second entry is "floating," i.e., not yet added
+## anywhere in  the tree. If the first entry is false, the node existed in the
+## tree earlier (and is presumed to have been added in some appropriate place).
+StructurePartMaybeFloating@ :=
```

Comment:
Please don't use the horrendous GAP "namespacing" feature. Instead, add an explicit `AUTODOC_` prefix to the function name (and consider using `BindGlobal` to assign it, to avoid user's from accidentally messing with it).